test/js_test.go

```diff
@@ -3844,7 +3845,9 @@ func TestJetStreamSubscribe_RateLimit(t *testing.T) {
 	}
 
 	// Change rate limit.
-	recvd := make(chan *nats.Msg)
+	// Make the receive channel able to possibly hold ALL messages, but
+	// we expect it to hold less due to rate limiting.
+	recvd := make(chan *nats.Msg, totalMsgs)
```

Comment:
Not sure exactly about the intent of this test, but since recvd channel was not buffered, it would block in the message callback trying to send message to it (since no one is reading from this channel).
Maybe you wanted to make sure that we don't put in the channel more than the rate limit? Anyway, feel free to modify this, but clearly the recvd channel cannot be unbuffered if no routine receives from it.

Reply:
Thanks, it should have been trying to buffer all messages. I used a channel there to avoid races when checking length with an array of messages.